src/Adapter/Product/ProductFieldValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShopException;
+use Product;
+
+/**
+ * Validates product field using legacy object model
+ */
+class ProductFieldValidator
```

Comment:
```suggestion
class ProductValidator
```

The `Field` suffix si not necessary, maybe you named it this way because of `CustomizationFieldValidator` except it is used for the `CustomizationField` object so it's not a suffix it's part of the Object name.

Reply:
I think I can make it ProductValidator and add methods like "validateProperty" and "validateLocalizedProperty" ?
As its in adapter, more methods can be added later depending on usecase :thinking: (something like validateProduct which maybe would validate all the fields)